server/controllers/getReferralCode/index.ts

```diff
@@ -19,6 +24,29 @@ const getReferralCode =
       address,
     );
 
+    let blacklistResult = null;
```

Comment:
Not critical. But it can be done in SELECT from user_kyc query like
```
SELECT 
uk.referral_code,
IIF(IFNULL(b.address, 'ok') == 'ok', 0, 1) blacklisted 
FROM user_kyc uk 
LEFT JOIN blacklist b ON (b.address = uk.address)
WHERE 
	uk.address = ?
LIMIT 1
```

Reply:
same in postKyc

Reply:
This way, we couldn't distinguish the case when kyc is not completed from the case when the address is in the blacklist. But perhaps it can still be done in one request

Reply:
you can distinguish this way
* query returned nothing = kyc not passed
* query returned ref_code and blacklisted=1 - user blacklisted
* query returned ref_code and blacklisted=0 - all good

Reply:
or you mean you need case when kyc not passed but user has been blacklisted?

Reply:
`query returned ref_code and blacklisted=1 ` - this case is not possible now cause we don't accept KYCs so dont generate referral codes for blacklisted addresses

Reply:
```
SELECT 
 f.address, 
IIF(IFNULL(b.address, 'ok') == 'ok', 0, 1) blacklisted, 
IIF(IFNULL(uk.address, 'ok') == 'ok', 0, 1) kyc_passed
FROM 
 (SELECT 'neutron1002fmu6uekkfrdldux7uz3u20nclrv7d7kq9wv' address ) f
 LEFT JOIN blacklist b ON (b.address = f.address)
 LEFT JOIN user_kyc uk ON (uk.address = f.address)
 ```